### PR TITLE
Added boolean and phrase queries

### LINKED ISSUE #27: Add support for phrase query
Query nodes are currently unsupported: https://github.com/Yelp/platypus/blob/0e3f63bed6788ce655976803c661af01f00505a2/src/main/java/org/apache/platypus/server/luceneserver/SearchHandler.java#L418

Reference code for phrase query: https://github.com/mikemccand/luceneserver/blob/c82373625ce9f92d2080a2f07c7d46cbd17e6ac1/src/java/org/apache/lucene/server/handlers/SearchHandler.java#L1138

### PR DESCRIPTION
**Pull request by sarthakn7**
Fixes #27 and #28 .

### CODE REVIEW

File: src/main/java/org/apache/platypus/server/luceneserver/QueryNodeMapper.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        }
+    }
+
+    private BooleanQuery getBooleanQuery(org.apache.platypus.server.grpc.BooleanQuery booleanQuery) {
+        BooleanQuery.Builder builder = new BooleanQuery.Builder()
+                .setMinimumNumberShouldMatch(booleanQuery.getMinimumNumberShouldMatch());
+
+        booleanQuery.getClausesList()
+                .forEach(clause -> builder
+                        .add(getQuery(clause.getQuery()), occurMapping.get(clause.getOccur()))
```
[the remark this answers is not in the document]
Comment (sarthakn7) [reply src/main/java/org/apache/platypus/server/luceneserver/QueryNodeMapper.java]: BooleanQuery.Builder in Lucene throws an exception if too many clauses are added:
https://github.com/apache/lucene-solr/blob/master/lucene/core/src/java/org/apache/lucene/search/BooleanQuery.java#L114

The max number of clauses is 1024 by default:
https://github.com/apache/lucene-solr/blob/master/lucene/core/src/java/org/apache/lucene/search/IndexSearcher.java#L91

Will this be enough, or do you think we need our own check ?

File: src/main/java/org/apache/platypus/server/luceneserver/QueryNodeMapper.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
+package org.apache.platypus.server.luceneserver;
```
[the remark this answers is not in the document]
Comment (sarthakn7) [reply src/main/java/org/apache/platypus/server/luceneserver/QueryNodeMapper.java]: Thanks, I've updated the PR with the license

[the remark this answers is not in the document]
Comment (sarthakn7) [reply src/main/java/org/apache/platypus/server/luceneserver/QueryNodeMapper.java]: Didn't change the commit message 🤦‍♂

File: src/main/java/org/apache/platypus/server/luceneserver/QueryNodeMapper.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (umeshdangat) [reply src/main/java/org/apache/platypus/server/luceneserver/QueryNodeMapper.java]: 👍

File: src/main/java/org/apache/platypus/server/luceneserver/QueryNodeMapper.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (umeshdangat) [reply src/main/java/org/apache/platypus/server/luceneserver/QueryNodeMapper.java]: I generally try to squash my commits into one before merging.

**Pull request merged**